include/mystery_gift.h

```diff
@@ -33,37 +38,153 @@ typedef struct MG_MON_DECO_TAG {
     int id;
 } MG_MON_DECO_TAG;
 
+typedef union {
+    MG_POKEMON_TAG pokemon;
+    Pokemon egg;
+    u32 item;
+    u16 ruleset[24];
+    int base_decoration;
+    MG_MON_DECO_TAG mon_decoration;
```

Comment:
Don't use snake_case for struct variables

Reply:
Also MG_MON_DECO_TAG to non-capitalized enum/struct name

Reply:
tbf this used to be an anonymous union inside `struct MysteryGift`. variant names are unchanged from that union.